libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -118,13 +118,16 @@ AP_RunCam::Menu AP_RunCam::_menus[RUNCAM_MAX_DEVICE_TYPES] = {
 	{ 6, { 3, 10, 2, 2, 8 }}, // Runcam 2 4K
 };
 
-AP_RunCam::AP_RunCam()
+AP_RunCam::AP_RunCam(AP_Camera &frontend, AP_Camera_Params &params, uint8_t instance)
+    : AP_Camera_Backend(frontend, params, instance)
 {
     AP_Param::setup_object_defaults(this, var_info);
-    if (_singleton != nullptr) {
+    if (_singleton != nullptr && _singleton->_instance == instance) {
```

Comment:
This looks very bad.  I mean, it's right there in the name...

You probably simply need to remove the singleton from AP_RunCam entirely.

Reply:
Right, but the thing isn't a singleton.  It's just the first instance.  This means we've got a significantly different meaning for the word "singleton" within the RunCam library to everywhere else in ArduPilot.

This can be done better in several different ways.  You can create specially-named methods on AP::runcam() - eg. `AP::camera().runcam_start_recording()`, and we could also look at polling the aux-switch state in the backend.  Could also just rename singleton to `_first_instance`.

... but I really can't ask you to do that in this PR, so I think we just put up with this pretty nasty discrepancy.

Reply:
Fair comment